hal/common/retarget.cpp

```diff
@@ -713,6 +713,39 @@ extern "C" void __env_unlock( struct _reent *_r )
 {
     __rtos_env_unlock(_r);
 }
+
+extern "C" int __cxa_guard_acquire (uint64_t *guard_object_p)
```

Comment:
The signature is a bit different for ARM aeabi, an `int*` is used  as parameter.
Same thing for `__cxa_guard_release` and `__cxa_guard_abort`.

More details here: http://infocenter.arm.com/help/topic/com.arm.doc.ihi0043d/IHI0043D_rtabi.pdf